dataduct/etl/etl_pipeline.py

```diff
@@ -71,21 +72,27 @@ class ETLPipeline(object):
     # easier
     DEFAULT_TOPIC_ARN = config.etl.get('DEFAULT_TOPIC_ARN', const.NONE)
     DEFAULT_TOPIC_ARN_SUCCESS = config.etl.get('DEFAULT_TOPIC_ARN_SUCCESS', const.NONE)
+    DEFAULT_TOPIC_ARN_ONLATE = config.etl.get('DEFAULT_TOPIC_ARN_ONLATE', const.NONE)
 
     def __init__(self, name, frequency='one-time', ec2_resource_config=None,
                  time_delta=None, emr_cluster_config=None, load_time=None,
                  topic_arn=None, max_retries=MAX_RETRIES, teardown=None,
-                 bootstrap=None, description=None, topic_arn_success=None):
+                 bootstrap=None, description=None, topic_arn_success=None, topic_arn_onlate=None, onlate_timeout=None, teardown_send_sns_late=True):
         """Constructor for the pipeline class
 
         Args:
             name (str): Name of the pipeline should be globally unique.
             frequency (enum): Frequency of the pipeline. Can be
             time_delta(timedelta): Duration to change the start time by
             emr_cluster_config(dict): Dictionary for emr config
-            topic_arn(str): sns alert to be used by the pipeline
+            topic_arn(str): sns alert to be used by the pipeline when it fails
             max_retries(int): number of retries for pipeline activities
             bootstrap(list of steps): bootstrap step definitions for resources
+            description(str): default None. Defines a description for the pipeline
+            topic_arn_success(str): default None. sns alert to be used by the pipeline when it succeeds
+            topic_arn_onlate(str): default None. sns alert to be used by the pipeline when it delays. onlate_timeout should be defined in order to trigger the alert.
+            onlate_timeout(str): default None. Time elapsed when the pipeline should consider "late", considering it's scheduled time. Value should represent one of the keys defined in dataduct/utils/constants.py:52.
+            teardown_send_sns_late(bool): default True. whether onlateaction should be triggered on teardown. If true, onlate_timeout should be defined in order to trigger the alert.
```

Comment:
I'm....confused by this parameter. Description states
```
whether onlateaction should be triggered on teardown. If true, onlate_timeout should be defined in order to trigger the alert.
```

I don't see where `teardown_send_sns_late` is attached to the teardown step anywhere in the code within this PR.

Additionally, given the description, I don't understand why this would default to True. Especially since `onlate_timeout` defaults to None

Came across this in https://github.com/EverFi/datafi/pull/334/files#diff-5e7f26a58e3e8e5d9e58e0ee24154822R11

Reply:
Since the onlate_timeout is set at pipeline level (not step) the idea is to use the teardown_send_sns_late to decide if the onlate should be attached to the teardown. This is useful for the case you want to attach the onlate alarm to a particular step, but do not want to attach to the teardown. This [line](https://github.com/EverFi/dataduct/blob/bc2094eec50574d4d11658f01b7d483ece2f0351/dataduct/etl/etl_pipeline.py#L279)  instantiates the alarm based on the flag, so then the alarm is sent as param to the etl_step when creating the teardown. See this [line](https://github.com/EverFi/dataduct/blob/bc2094eec50574d4d11658f01b7d483ece2f0351/dataduct/etl/etl_pipeline.py#L619)
Hope this clarify your confusion, otherwise we can schedule a call.

Regarding the default value, I set it to false. In case we need to attach onlate to teardown, onlate_timeout + teardown_send_sns_late=True should be set on yml.

Reply:
Gotcha, makes sense, thanks for the clarification 👍 

Thanks for fixing up docs on the default value.